frontend/src/components/pages/Account.tsx

```diff
@@ -1,42 +1,210 @@
+  const [formErrors, setFormErrors] = useState<{
+    businessName: string;
+    firstName: string;
+    lastName: string;
+    phoneNumber: string;
+  }>({
+    businessName: "",
+    firstName: "",
+    lastName: "",
+    phoneNumber: "",
+  });
+
+  const navigateToDashboard = () => {
+    history.push(Routes.DASHBOARD_PAGE);
+  };
+
+  const changeEditMode = () => {
+    setIsEditing(!isEditing);
+  };
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement> | string,
+    name: string,
+  ) => {
+    setForm({ target: { name, value: e } });
+    if (name === "businessName") {
+      setBusinessName(e.toString());
+    }
+  };
+
+  const discardChanges = () => {
+    // reset all form values (except email) and businessName
+    // and return to not editing mode
+    formValues.firstName = authenticatedUser!.firstName;
+    formValues.lastName = authenticatedUser!.lastName;
+    formValues.phoneNumber = authenticatedUser!.phoneNumber;
+    setBusinessName(donor!.businessName);
+    setIsEditing(false);
+    onClose();
+    setFormErrors({
+      businessName: "",
+      firstName: "",
+      lastName: "",
+      phoneNumber: "",
+    });
+  };
+
+  const validateForm = () => {
+    const newErrors = {
+      businessName: "",
+      firstName: "",
+      lastName: "",
+      phoneNumber: "",
+    };
+
+    let isValid = true;
+
+    if (!businessName) {
+      isValid = false;
+      newErrors.businessName = ErrorMessages.requiredField;
+    }
+    if (!formValues.firstName) {
+      isValid = false;
+      newErrors.firstName = ErrorMessages.requiredField;
+    }
+    if (!formValues.lastName) {
+      isValid = false;
+      newErrors.lastName = ErrorMessages.requiredField;
+    }
+    if (!formValues.phoneNumber) {
+      isValid = false;
+      newErrors.phoneNumber = ErrorMessages.requiredField;
+    }
+
+    setFormErrors(newErrors);
+    return isValid;
+  };
+
+  const onSubmitClick = async () => {
+    if (!validateForm()) return;
+
+    // format user request data object
+    const userData = {
+      id: authenticatedUser!.id,
+      role: authenticatedUser!.role,
+      ...formValues,
+    };
+    const updatedUser = await UserAPIClient.updateUserById(
+      authenticatedUser!.id,
+      {
+        userData,
+      },
+    );
+    await DonorAPIClient.updateDonorById(authenticatedUser!.id, {
+      businessName,
+    });
+
+    setIsEditing(false);
+
+    // update authenticatedUser and local storage to reflect changes
+    const user = {
+      accessToken: authenticatedUser!.accessToken,
+      ...updatedUser,
+    };
+    setAuthenticatedUser(user);
+
+    const keys = Object.keys(user);
+    const values = Object.values(user);
+    for (let i = 0; i < keys.length; i += 1) {
+      if (
+        keys[i] === "firstName" ||
+        keys[i] === "lastName" ||
+        keys[i] === "phoneNumber"
+      ) {
+        setLocalStorageObjProperty(AUTHENTICATED_USER_KEY, keys[i], values[i]);
+      }
+    }
+  };
```

Comment:
```suggestion
    for (const [key, value] in Object.entries(user)){
        if(key === "firstName" || key === "lastName" || key === "phoneNumber"){
            setLocalStorageObjProperty(AUTHENTICATED_USER_KEY, key, value);
        }
    }
    
```

Reply:
This gives the error `src/components/pages/Account.tsx
  Line 175:5:  for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries}, and iterate over the resulting array  no-restricted-syntax`